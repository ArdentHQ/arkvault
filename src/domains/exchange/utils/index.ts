export * from "./assertions";
export * from "./errors";
