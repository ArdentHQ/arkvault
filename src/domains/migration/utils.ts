import { MigrationTransactionStatus } from "@/domains/migration/migration.contracts";

export const getIcon = (status: MigrationTransactionStatus) => {
	if (status === MigrationTransactionStatus.Confirmed) {
		return {
			color: "text-theme-success-600",
			name: "CircleCheckMark",
		};
	}

	return {
		color: "text-theme-warning-300",
		name: "Clock",
	};
};
