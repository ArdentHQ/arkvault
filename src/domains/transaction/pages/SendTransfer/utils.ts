export const getFeeType = (numberOfRecipients: number): "multiPayment" | "transfer" =>
	numberOfRecipients > 1 ? "multiPayment" : "transfer";
