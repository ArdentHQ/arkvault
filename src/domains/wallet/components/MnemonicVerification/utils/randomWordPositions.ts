import { sample } from "@ardenthq/sdk-helpers";

const randomWordPositions = (length: number): number[] => {
	const positions: number[] = [...Array.from({ length }).keys()];
	const result: number[] = [];

	while (result.length < 3) {
		const randomNumber = sample(positions) + 1;

		if (result.includes(randomNumber)) {
			continue;
		}

		result.push(randomNumber);
	}

	return result.sort((a, b) => a - b);
};

export { randomWordPositions };
