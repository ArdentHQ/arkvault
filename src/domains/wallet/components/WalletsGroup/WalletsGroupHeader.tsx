import React from "react";
import { useTranslation } from "react-i18next";
import { useBreakpoint } from "@/app/hooks";
import {
	GroupNetworkIcon,
	GroupNetworkName,
	GroupNetworkTotal,
	GroupSkeleton,
	LabelledText,
} from "@/domains/wallet/components/WalletsGroup/WalletsGroup.blocks";
import {
	WalletsGroupHeaderProperties,
	WalletsGroupHeaderSkeletonProperties,
} from "@/domains/wallet/components/WalletsGroup/WalletsGroup.contracts";
import { AccordionHeader, AccordionHeaderSkeletonWrapper } from "@/app/components/Accordion";
import { Skeleton } from "@/app/components/Skeleton";

export const WalletsGroupHeader: React.VFC<WalletsGroupHeaderProperties> = ({
	wallets,
	network,
	onClick,
	isExpanded,
	maxWidthReferences,
	className,
	isSyncing,
}) => (
	<AccordionHeader isExpanded={isExpanded} onClick={onClick} data-testid="WalletsGroupHeader" className={className}>
		<GroupNetworkIcon network={network} isGroupExpanded={isExpanded} />
		<GroupNetworkName network={network} />
		<GroupNetworkTotal
			network={network}
			wallets={wallets}
			maxWidthReferences={maxWidthReferences}
			noBorder={!onClick}
			isSyncing={isSyncing}
		/>
	</AccordionHeader>
);

export const WalletsGroupHeaderSkeleton: React.VFC<WalletsGroupHeaderSkeletonProperties> = ({
	isPlaceholder = false,
}) => {
	const { t } = useTranslation();

	const { isXs } = useBreakpoint();

	const labelClassname = "text-lg text-theme-secondary-600 dark:text-theme-secondary-700";
	const labelNotAvailable = t("COMMON.NOT_AVAILABLE");

	return (
		<AccordionHeaderSkeletonWrapper data-testid="WalletsGroupHeaderSkeleton">
			<div className="flex h-11 w-11 shrink-0 flex-row items-center justify-center -space-x-1 rounded-xl bg-theme-background bg-clip-padding text-theme-secondary-100 ring-1 ring-theme-secondary-300 dark:bg-theme-background dark:text-theme-secondary-700 dark:ring-theme-secondary-800">
				{isPlaceholder ? <></> : <Skeleton width={20} height={20} />}
			</div>
			<div className="grow justify-between space-x-3 py-3 px-3">
				<h1
					className="mb-0 text-lg text-theme-secondary-500 dark:text-theme-secondary-700"
					data-testid="header__title"
				>
					{isPlaceholder ? (
						t("COMMON.CRYPTOASSET")
					) : (
						<div className="flex h-7 items-center">
							<Skeleton width={150} height={18} />
						</div>
					)}
				</h1>
			</div>

			{isXs && !isPlaceholder && (
				<div className="flex h-10 flex-col items-end justify-between">
					<Skeleton width={90} height={14} />
					<Skeleton width={50} height={12} />
				</div>
			)}

			{!isXs && (
				<div className="flex space-x-3">
					<LabelledText label={t("COMMON.WALLETS")}>
						<span className={labelClassname}>
							{isPlaceholder ? labelNotAvailable : <GroupSkeleton width={60} />}
						</span>
					</LabelledText>

					<LabelledText label={t("COMMON.TOTAL_BALANCE")}>
						<span className={labelClassname}>
							{isPlaceholder ? labelNotAvailable : <GroupSkeleton width={160} />}
						</span>
					</LabelledText>

					<LabelledText label={t("COMMON.TOTAL_CURRENCY")} className="hidden lg:flex">
						<span className={labelClassname}>
							{isPlaceholder ? labelNotAvailable : <GroupSkeleton width={130} />}
						</span>
					</LabelledText>
				</div>
			)}
		</AccordionHeaderSkeletonWrapper>
	);
};
