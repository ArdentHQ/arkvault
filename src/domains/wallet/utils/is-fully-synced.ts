import { Contracts } from "@payvo/sdk-profiles";

export const isFullySynced = (wallet: Contracts.IReadWriteWallet): boolean =>
	wallet.hasBeenFullyRestored() && (wallet.hasSyncedWithNetwork() || wallet.balance() === 0);
