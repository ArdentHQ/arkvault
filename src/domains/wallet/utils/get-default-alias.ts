import { Networks } from "@ardenthq/sdk";
import { Contracts } from "@ardenthq/sdk-profiles";
import { networkDisplayName } from "@/utils/network-utils";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${networkDisplayName(network)} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let counter = sameCoinWallets.length;

	if (counter === 0) {
		counter = 1;
	}

	while (profile.wallets().findByAlias(makeAlias(counter))) {
		counter++;
	}

	return makeAlias(counter);
};
