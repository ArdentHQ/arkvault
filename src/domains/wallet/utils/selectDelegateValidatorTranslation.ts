import { Networks } from "@ardenthq/sdk";
import { isMainsailNetwork } from "@/utils/network-utils";

interface Properties<T extends string | React.ReactNode> {
	network: Networks.Network;
	validatorStr: T;
	delegateStr: T;
}

export const selectDelegateValidatorTranslation = <T extends string | React.ReactNode>({ network, validatorStr, delegateStr }: Properties<T>): T => isMainsailNetwork(network) ? validatorStr : delegateStr;



