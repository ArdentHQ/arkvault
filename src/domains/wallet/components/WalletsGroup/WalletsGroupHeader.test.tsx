import React from "react";
import { WalletsGroupHeaderSkeleton } from "./WalletsGroupHeader";
import { render, renderResponsive } from "@/utils/testing-library";

describe("WalletsGroupHeaderSkeleton", () => {
	it("should render", () => {
		const { asFragment } = render(<WalletsGroupHeaderSkeleton />);

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render responsive", () => {
		const { asFragment } = renderResponsive(<WalletsGroupHeaderSkeleton />, "xs");

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render as placeholder", () => {
		const { asFragment } = render(<WalletsGroupHeaderSkeleton placeholder />);

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render responsive as placeholder", () => {
		const { asFragment } = renderResponsive(<WalletsGroupHeaderSkeleton placeholder />, "xs");

		expect(asFragment()).toMatchSnapshot();
	});
});
