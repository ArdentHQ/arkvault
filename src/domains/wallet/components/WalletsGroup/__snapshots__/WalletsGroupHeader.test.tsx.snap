// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WalletsGroupHeaderSkeleton > should render 1`] = `
<DocumentFragment>
  <div
    class="flex h-20 items-center px-8 py-4 md:px-4"
    data-testid="WalletsGroupHeaderSkeleton"
  >
    <div
      class="flex h-11 w-11 shrink-0 flex-row items-center justify-center -space-x-1 rounded-xl bg-theme-background bg-clip-padding text-theme-secondary-100 ring-1 ring-theme-secondary-300 dark:bg-theme-background dark:text-theme-secondary-700 dark:ring-theme-secondary-800"
    >
      <span
        aria-busy="true"
        aria-live="polite"
        class="flex w-auto max-w-full items-center overflow-hidden leading-none"
      >
        <span
          class="react-loading-skeleton"
          style="width: 20px; height: 20px;"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
    <div
      class="grow justify-between space-x-3 px-3 py-3"
    >
      <h1
        class="mb-0 text-lg text-theme-secondary-700 dark:text-theme-secondary-700"
        data-testid="header__title"
      >
        <div
          class="flex h-7 items-center"
        >
          <span
            aria-busy="true"
            aria-live="polite"
            class="flex w-auto max-w-full items-center overflow-hidden leading-none"
          >
            <span
              class="react-loading-skeleton"
              style="width: 150px; height: 18px;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
      </h1>
    </div>
    <div
      class="flex space-x-3"
    >
      <div
        class="flex flex-col space-y-1 border-r border-theme-secondary-300 pr-4 text-right font-semibold dark:border-theme-secondary-800"
      >
        <span
          class="whitespace-nowrap text-sm text-theme-secondary-700 dark:text-theme-secondary-500"
        >
          Wallets
        </span>
        <span
          class="text-lg text-theme-secondary-600 dark:text-theme-secondary-700"
        >
          <div
            class="flex items-center justify-end sm:h-6"
          >
            <span
              aria-busy="true"
              aria-live="polite"
              class="flex w-auto max-w-full items-center overflow-hidden leading-none"
            >
              <span
                class="react-loading-skeleton sm:h-4.5"
                style="width: 60px;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
        </span>
      </div>
      <div
        class="flex flex-col space-y-1 border-r border-theme-secondary-300 pr-4 text-right font-semibold dark:border-theme-secondary-800"
      >
        <span
          class="whitespace-nowrap text-sm text-theme-secondary-700 dark:text-theme-secondary-500"
        >
          Total Balance
        </span>
        <span
          class="text-lg text-theme-secondary-600 dark:text-theme-secondary-700"
        >
          <div
            class="flex items-center justify-end sm:h-6"
          >
            <span
              aria-busy="true"
              aria-live="polite"
              class="flex w-auto max-w-full items-center overflow-hidden leading-none"
            >
              <span
                class="react-loading-skeleton sm:h-4.5"
                style="width: 160px;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
        </span>
      </div>
      <div
        class="flex flex-col space-y-1 border-r border-theme-secondary-300 pr-4 text-right font-semibold dark:border-theme-secondary-800 hidden lg:flex"
      >
        <span
          class="whitespace-nowrap text-sm text-theme-secondary-700 dark:text-theme-secondary-500"
        >
          Total Currency
        </span>
        <span
          class="text-lg text-theme-secondary-600 dark:text-theme-secondary-700"
        >
          <div
            class="flex items-center justify-end sm:h-6"
          >
            <span
              aria-busy="true"
              aria-live="polite"
              class="flex w-auto max-w-full items-center overflow-hidden leading-none"
            >
              <span
                class="react-loading-skeleton sm:h-4.5"
                style="width: 130px;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
        </span>
      </div>
    </div>
    <div
      class="flex h-full items-center border-theme-secondary-300 pl-4 dark:border-theme-secondary-800 dark:group-hover:border-theme-secondary-700"
    >
      <div
        class="flex h-8 w-8 content-center items-center justify-center rounded-lg bg-theme-background text-theme-secondary-500 ring-1 ring-theme-secondary-300 dark:text-theme-secondary-700 dark:ring-theme-secondary-800"
      >
        <div>
          <div
            style="height: 10px; width: 10px;"
          >
            <svg
              id="chevron-down-small"
              style="height: 100%; width: 100%;"
              viewBox="0 0 10 10"
              x="0"
              xml:space="preserve"
              xmlns="http://www.w3.org/2000/svg"
              y="0"
            >
              <path
                d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`WalletsGroupHeaderSkeleton > should render as placeholder 1`] = `
<DocumentFragment>
  <div
    class="flex h-20 items-center px-8 py-4 md:px-4"
    data-testid="WalletsGroupHeaderSkeleton"
  >
    <div
      class="flex h-11 w-11 shrink-0 flex-row items-center justify-center -space-x-1 rounded-xl bg-theme-background bg-clip-padding text-theme-secondary-100 ring-1 ring-theme-secondary-300 dark:bg-theme-background dark:text-theme-secondary-700 dark:ring-theme-secondary-800"
    >
      <span
        aria-busy="true"
        aria-live="polite"
        class="flex w-auto max-w-full items-center overflow-hidden leading-none"
      >
        <span
          class="react-loading-skeleton"
          style="width: 20px; height: 20px;"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
    <div
      class="grow justify-between space-x-3 px-3 py-3"
    >
      <h1
        class="mb-0 text-lg text-theme-secondary-700 dark:text-theme-secondary-700"
        data-testid="header__title"
      >
        <div
          class="flex h-7 items-center"
        >
          <span
            aria-busy="true"
            aria-live="polite"
            class="flex w-auto max-w-full items-center overflow-hidden leading-none"
          >
            <span
              class="react-loading-skeleton"
              style="width: 150px; height: 18px;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
      </h1>
    </div>
    <div
      class="flex space-x-3"
    >
      <div
        class="flex flex-col space-y-1 border-r border-theme-secondary-300 pr-4 text-right font-semibold dark:border-theme-secondary-800"
      >
        <span
          class="whitespace-nowrap text-sm text-theme-secondary-700 dark:text-theme-secondary-500"
        >
          Wallets
        </span>
        <span
          class="text-lg text-theme-secondary-600 dark:text-theme-secondary-700"
        >
          <div
            class="flex items-center justify-end sm:h-6"
          >
            <span
              aria-busy="true"
              aria-live="polite"
              class="flex w-auto max-w-full items-center overflow-hidden leading-none"
            >
              <span
                class="react-loading-skeleton sm:h-4.5"
                style="width: 60px;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
        </span>
      </div>
      <div
        class="flex flex-col space-y-1 border-r border-theme-secondary-300 pr-4 text-right font-semibold dark:border-theme-secondary-800"
      >
        <span
          class="whitespace-nowrap text-sm text-theme-secondary-700 dark:text-theme-secondary-500"
        >
          Total Balance
        </span>
        <span
          class="text-lg text-theme-secondary-600 dark:text-theme-secondary-700"
        >
          <div
            class="flex items-center justify-end sm:h-6"
          >
            <span
              aria-busy="true"
              aria-live="polite"
              class="flex w-auto max-w-full items-center overflow-hidden leading-none"
            >
              <span
                class="react-loading-skeleton sm:h-4.5"
                style="width: 160px;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
        </span>
      </div>
      <div
        class="flex flex-col space-y-1 border-r border-theme-secondary-300 pr-4 text-right font-semibold dark:border-theme-secondary-800 hidden lg:flex"
      >
        <span
          class="whitespace-nowrap text-sm text-theme-secondary-700 dark:text-theme-secondary-500"
        >
          Total Currency
        </span>
        <span
          class="text-lg text-theme-secondary-600 dark:text-theme-secondary-700"
        >
          <div
            class="flex items-center justify-end sm:h-6"
          >
            <span
              aria-busy="true"
              aria-live="polite"
              class="flex w-auto max-w-full items-center overflow-hidden leading-none"
            >
              <span
                class="react-loading-skeleton sm:h-4.5"
                style="width: 130px;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
        </span>
      </div>
    </div>
    <div
      class="flex h-full items-center border-theme-secondary-300 pl-4 dark:border-theme-secondary-800 dark:group-hover:border-theme-secondary-700"
    >
      <div
        class="flex h-8 w-8 content-center items-center justify-center rounded-lg bg-theme-background text-theme-secondary-500 ring-1 ring-theme-secondary-300 dark:text-theme-secondary-700 dark:ring-theme-secondary-800"
      >
        <div>
          <div
            style="height: 10px; width: 10px;"
          >
            <svg
              id="chevron-down-small"
              style="height: 100%; width: 100%;"
              viewBox="0 0 10 10"
              x="0"
              xml:space="preserve"
              xmlns="http://www.w3.org/2000/svg"
              y="0"
            >
              <path
                d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`WalletsGroupHeaderSkeleton > should render responsive 1`] = `
<DocumentFragment>
  <div
    class="flex h-20 items-center px-8 py-4 md:px-4"
    data-testid="WalletsGroupHeaderSkeleton"
  >
    <div
      class="flex h-11 w-11 shrink-0 flex-row items-center justify-center -space-x-1 rounded-xl bg-theme-background bg-clip-padding text-theme-secondary-100 ring-1 ring-theme-secondary-300 dark:bg-theme-background dark:text-theme-secondary-700 dark:ring-theme-secondary-800"
    >
      <span
        aria-busy="true"
        aria-live="polite"
        class="flex w-auto max-w-full items-center overflow-hidden leading-none"
      >
        <span
          class="react-loading-skeleton"
          style="width: 20px; height: 20px;"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
    <div
      class="grow justify-between space-x-3 px-3 py-3"
    >
      <h1
        class="mb-0 text-lg text-theme-secondary-700 dark:text-theme-secondary-700"
        data-testid="header__title"
      >
        <div
          class="flex h-7 items-center"
        >
          <span
            aria-busy="true"
            aria-live="polite"
            class="flex w-auto max-w-full items-center overflow-hidden leading-none"
          >
            <span
              class="react-loading-skeleton"
              style="width: 150px; height: 18px;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
      </h1>
    </div>
    <div
      class="flex h-10 flex-col items-end justify-between"
    >
      <span
        aria-busy="true"
        aria-live="polite"
        class="flex w-auto max-w-full items-center overflow-hidden leading-none"
      >
        <span
          class="react-loading-skeleton"
          style="width: 90px; height: 14px;"
        >
          ‌
        </span>
        <br />
      </span>
      <span
        aria-busy="true"
        aria-live="polite"
        class="flex w-auto max-w-full items-center overflow-hidden leading-none"
      >
        <span
          class="react-loading-skeleton"
          style="width: 50px; height: 12px;"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
    <div
      class="flex h-full items-center border-theme-secondary-300 pl-4 dark:border-theme-secondary-800 dark:group-hover:border-theme-secondary-700"
    >
      <div
        class="flex h-8 w-8 content-center items-center justify-center rounded-lg bg-theme-background text-theme-secondary-500 ring-1 ring-theme-secondary-300 dark:text-theme-secondary-700 dark:ring-theme-secondary-800"
      >
        <div>
          <div
            style="height: 10px; width: 10px;"
          >
            <svg
              id="chevron-down-small"
              style="height: 100%; width: 100%;"
              viewBox="0 0 10 10"
              x="0"
              xml:space="preserve"
              xmlns="http://www.w3.org/2000/svg"
              y="0"
            >
              <path
                d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`WalletsGroupHeaderSkeleton > should render responsive as placeholder 1`] = `
<DocumentFragment>
  <div
    class="flex h-20 items-center px-8 py-4 md:px-4"
    data-testid="WalletsGroupHeaderSkeleton"
  >
    <div
      class="flex h-11 w-11 shrink-0 flex-row items-center justify-center -space-x-1 rounded-xl bg-theme-background bg-clip-padding text-theme-secondary-100 ring-1 ring-theme-secondary-300 dark:bg-theme-background dark:text-theme-secondary-700 dark:ring-theme-secondary-800"
    >
      <span
        aria-busy="true"
        aria-live="polite"
        class="flex w-auto max-w-full items-center overflow-hidden leading-none"
      >
        <span
          class="react-loading-skeleton"
          style="width: 20px; height: 20px;"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
    <div
      class="grow justify-between space-x-3 px-3 py-3"
    >
      <h1
        class="mb-0 text-lg text-theme-secondary-700 dark:text-theme-secondary-700"
        data-testid="header__title"
      >
        <div
          class="flex h-7 items-center"
        >
          <span
            aria-busy="true"
            aria-live="polite"
            class="flex w-auto max-w-full items-center overflow-hidden leading-none"
          >
            <span
              class="react-loading-skeleton"
              style="width: 150px; height: 18px;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
      </h1>
    </div>
    <div
      class="flex h-10 flex-col items-end justify-between"
    >
      <span
        aria-busy="true"
        aria-live="polite"
        class="flex w-auto max-w-full items-center overflow-hidden leading-none"
      >
        <span
          class="react-loading-skeleton"
          style="width: 90px; height: 14px;"
        >
          ‌
        </span>
        <br />
      </span>
      <span
        aria-busy="true"
        aria-live="polite"
        class="flex w-auto max-w-full items-center overflow-hidden leading-none"
      >
        <span
          class="react-loading-skeleton"
          style="width: 50px; height: 12px;"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
    <div
      class="flex h-full items-center border-theme-secondary-300 pl-4 dark:border-theme-secondary-800 dark:group-hover:border-theme-secondary-700"
    >
      <div
        class="flex h-8 w-8 content-center items-center justify-center rounded-lg bg-theme-background text-theme-secondary-500 ring-1 ring-theme-secondary-300 dark:text-theme-secondary-700 dark:ring-theme-secondary-800"
      >
        <div>
          <div
            style="height: 10px; width: 10px;"
          >
            <svg
              id="chevron-down-small"
              style="height: 100%; width: 100%;"
              viewBox="0 0 10 10"
              x="0"
              xml:space="preserve"
              xmlns="http://www.w3.org/2000/svg"
              y="0"
            >
              <path
                d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
